stripe/src/main/java/com/stripe/android/view/MaskedCardView.java

```diff
@@ -0,0 +1,270 @@
+        TEMPLATE_RESOURCE_MAP.put(Card.JCB, R.drawable.ic_jcb_template_32);
+        TEMPLATE_RESOURCE_MAP.put(Card.MASTERCARD, R.drawable.ic_mastercard_template_32);
+        TEMPLATE_RESOURCE_MAP.put(Card.VISA, R.drawable.ic_visa_template_32);
+        TEMPLATE_RESOURCE_MAP.put(Card.UNKNOWN, R.drawable.ic_unknown);
+    }
+
+    public MaskedCardView(Context context) {
+        super(context);
+        init();
+    }
+
+    public MaskedCardView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        init();
+    }
+
+    public MaskedCardView(Context context, AttributeSet attrs, int defStyle) {
+        super(context, attrs, defStyle);
+        init();
+    }
+
+    /**
+     * Set the card data displayed using a {@link Card} object. Note that
+     * the full number will not be accessed here.
+     *
+     * @param card the {@link Card} to be partially displayed
+     */
+    public void setCard(@NonNull Card card) {
+        mCardBrand = card.getBrand();
+        mLast4 = card.getLast4();
+        updateBrandIcon();
+        updateCardInformation();
+    }
+
+    /**
+     * Set the card data displayed using a {@link SourceCardData} object.
+     *
+     * @param sourceCardData the {@link SourceCardData} to be partially displayed
+     */
+    public void setSourceCardData(@NonNull SourceCardData sourceCardData) {
+        mCardBrand = sourceCardData.getBrand();
+        mLast4 = sourceCardData.getLast4();
+        updateBrandIcon();
+        updateCardInformation();
+    }
+
+    /**
+     * Set the card data displayed using a {@link CustomerSource} object. If the object
+     * is not a {@link Source} object that represents a card and the object returns {@code null}
+     * from its {@link CustomerSource#asCard()} method, then no values will be set on this control.
+     *
+     * @param customerSource the {@link CustomerSource} to be partially displayed
+     */
+    public void setCustomerSource(@NonNull CustomerSource customerSource) {
+        Source source = customerSource.asSource();
+        if (source != null
+                && source.getSourceTypeModel() != null
+                && Source.CARD.equals(source.getType())
+                && source.getSourceTypeModel() instanceof SourceCardData) {
+            SourceCardData sourceCardData = (SourceCardData) source.getSourceTypeModel();
+            setSourceCardData(sourceCardData);
+            return;
+        }
+
+        Card card = customerSource.asCard();
+        if (card != null) {
+            setCard(card);
+        }
+    }
+
+    /**
+     * @return whether or not this view is displaying as selected
+     */
+    public boolean isHighlighted() {
+        return mIsHighlighted;
+    }
+
+    /**
+     * @param isHighlighted whether or not this view should display in selected mode
+     */
+    public void setHighlighted(boolean isHighlighted) {
+        mIsHighlighted = isHighlighted;
```

Comment:
should we add an animation to this state change? the color change feels a little abrupt.